File: android/app/src/main/kotlin/net/chikach/submon/mch/ActionMethodChannelHandler.kt
package net.chikach.submon.mch

import io.flutter.embedding.engine.FlutterEngine
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import net.chikach.submon.METHOD_CHANNEL_ACTION

class ActionMethodChannelHandler : MethodChannel.MethodCallHandler {
    var pendingAction: Map<String, Any?>? = null

    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        when (call.method) {
            "getPendingAction" -> {
                result.success(pendingAction)
                pendingAction = null
            }
            else -> result.notImplemented()
        }
    }

    fun invokeIntentAction(flutterEngine: FlutterEngine?) {
        if (flutterEngine != null && pendingAction != null) {
            val mc =
                MethodChannel(flutterEngine.dartExecutor.binaryMessenger, METHOD_CHANNEL_ACTION)
            mc.invokeMethod(pendingAction!!["actionName"] as String, pendingAction!!["arguments"])
            pendingAction = null
        }
    }
}

File: ios/Runner/MethodChannelHandler/ActionMethodChannelHandler.swift
import Foundation
import Flutter

class ActionMethodChannelHandler {
    let methodChannel: FlutterMethodChannel
    let viewController: FlutterViewController
    
    init(viewController: FlutterViewController) {
        self.methodChannel = FlutterMethodChannel(name: "net.chikach.submon/action", binaryMessenger: viewController.binaryMessenger)
        self.viewController = viewController
    }
    
    func register() {
        methodChannel.setMethodCallHandler(handler)
    }
    
    private func handler(call: FlutterMethodCall, result: @escaping FlutterResult) {
//        let args = call.arguments as! [String: String]
        
        switch call.method {
        case "getPendingAction":
            result(nil)
        default:
            result(FlutterMethodNotImplemented)
        }
    }
    
    func invokeAction(actionName: String, arguments: [String: String]) {
        switch (actionName) {
        case "openCreateNewPage":
            methodChannel.invokeMethod(actionName, arguments: nil)
            break
        case "openSubmissionDetailPage":
            methodChannel.invokeMethod(actionName, arguments: [
                "submissionId": arguments["submissionId"]
            ])
        case "openFocusTimerPage":
            methodChannel.invokeMethod(actionName, arguments: [
                "digestiveId": arguments["digestiveId"]
            ])
            break
        default: break
        }
    }
}
